Sharpasonne/Engine.cs

```diff
@@ -15,17 +15,28 @@ public class Engine : IEngine
     {
         public Board Board { get; } = new Board();
 
-        public IImmutableDictionary<Type, IImmutableList<IRule<IGameAction>>> Rules { get; }
-            = ImmutableDictionary<Type, IImmutableList<IRule<IGameAction>>>.Empty;
+        /// <summary>
+        /// Players collection for managing player stats and turns.
+        /// </summary>
+        public Players Players { get; }
 
         /// <summary>
-        /// 
+        /// 1-index number of the player who's turn it is to play an action.
         /// </summary>
+        public int CurrentPlayerTurn { get; } = 1;
+
+        public IImmutableDictionary<Type, IImmutableList<IRule<IGameAction>>> Rules { get; }
+            = ImmutableDictionary<Type, IImmutableList<IRule<IGameAction>>>.Empty;
+
+        /// <summary>Attempts to create a Game engine.</summary>
         /// <param name="gameActions"></param>
         /// <param name="rules">Must provide list for every action to be used by
         /// Perform.</param>
-        public Engine(
-            [NotNull] IImmutableDictionary<Type, IImmutableList<IRule<IGameAction>>> rules)
+        /// <param name="gameActions"></param>
+        public static Option<Engine, Exception> Create(
+            [NotNull] IImmutableQueue<IGameAction>                                   gameActions,
+            [NotNull] IImmutableDictionary<Type, IImmutableList<IRule<IGameAction>>> rules,
```

Comment:
I know this was already called this, but the name is not representative.

Reply:
Why is the name, Create, not representative of a factory method?

Reply:
Sorry, I was talking about "rules"!

Reply:
Ah, true, again lets open another issue for after this 👍